docs/release/controllers.md

```diff
@@ -0,0 +1,178 @@
+A deprecated alpha version is subject to removal after a three months period.
+
+An alpha API is introduced when its proposal reaches the  `implementable` phase in the
+[Flux RFC process](https://github.com/fluxcd/flux2/tree/main/rfcs).
+We encourage users to try out the alpha APIs and provide feedback which is extremely
+valuable during early stages of development.
+
+### Beta version
+
+A beta version API e.g. `v2beta1` is considered well tested and safe to be used.
+
+The schema of objects may change in incompatible ways in a subsequent beta or stable API version.
+The Flux Custom Resources may require editing after a CRD update for which migration instructions will be
+provided as part of the controller changelog.
+
+A beta version API becomes deprecated once a subsequent beta or stable API version is released. 
+A deprecated beta version is subject to removal after a six months period.
+
+### Stable version
+
+A stable version API e.g. `v2` is considered feature complete.
+
+Any changes to the object schema do not require editing or re-creating of Flux Custom Resources.
+Schema fields can't be removed, only new fields can be added with a default value that
+doesn't affect the controller's current behaviour.
+
+A stable version API becomes deprecated once a subsequent stable version is released.
+Stable API versions are not subject to removal in any future release of a controller major version.
+
+In effect, this means that for as long as Flux `v2` is being maintained, all the stable API versions 
+will be supported.
+
+## Controller versioning
+
+The Flux controllers and their Go API packages are released by following the
+[Go module version numbering](https://go.dev/doc/modules/version-numbers) conventions:
+
+- `vX.Y.Z-RC.W` release candidates e.g. `v1.0.0-RC.1`
+- `vX.Y.Z` stable releases e.g. `v1.0.0`
+
+The API versioning and controller versioning are indirectly related. For example,
+a source-controller minor release `v1.1.0` can introduce a new API version
+`v1beta1` for a Kind `XRepository` in the `source.toolkit.fluxcd.io` group.
+
+### Release candidates
+
+Release candidates are intended for testing new features or improvements before a final release.
+
+In most cases, a maintainer will publish a release candidate of a controller for Flux users
+to tests it on their staging clusters. Release candidates are not meant to be deployed in production
+unless advised to do so by a maintainer.
+
+### Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security
+vulnerabilities or fixes to severe problems with no workaround.
+
+Patch releases do not contain breaking changes, feature additions or any type of user-facing changes.
+If a CVE fix requires a breaking change, then a minor release will provide the fix.
+
+We expect users to be running the latest patch release of a given minor release as soon as the
+controller release is included in a Flux patch release.
```

Comment:
Maybe?
```suggestion
controller release is included in a Flux CLI patch release.
```

Reply:
It's a Flux release which has nothing to do with the CLI binary. People can setup Flux to update itself from `ghcr.io/fluxcd/flux-manifests`, or they can use Flux TF provider or even the community maintained Helm charts.